### PR TITLE
add Lua analyzer

### PR DESCRIPTION
**Pull request by 3van**
Based off of the [Lua 5.3 Reference Manual](http://www.lua.org/manual/5.3/manual.html).
I ran it against our (crazy, massively large) codebase and didn't see it trip up on any of our Lua.
I signed the OCA *forever* ago when I used to contribute to the ndbcluster engine on MySQL; not sure if I need to sign anything else for this, but let me know if I do.

### CODE REVIEW

**Comment by vladak** [comment]
Nice ! would it be possible to add some tests ?

**Comment by 3van** [comment]
Sure. Let me see if I can whip up a representative Lua sample file.

File: src/org/opensolaris/opengrok/analysis/lua/LuaSymbolTokenizer.lex [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
+ */
+
+/*
+ * Get Golang symbols - ignores comments, strings, keywords
```
Comment (tarzanek) [inline src/org/opensolaris/opengrok/analysis/lua/LuaSymbolTokenizer.lex]: get LUA sumbols? ;)
copy paste ? :-D

**Comment by tarzanek** [comment]
@3van any news?
btw. I don't have a problem to integrate (don't want to hold this up) even without tests, if you promise you will add them in next pull req
so if you will fix the typo, I'll happily merge (and I am still figuring out if your mysql OCA can be used, but I hope it can)

**Comment by tarzanek** [comment]
@3van confirming your OCA, so basically we're just waiting for you :)
tia
L

File: src/org/opensolaris/opengrok/analysis/lua/LuaSymbolTokenizer.lex (hunk 1 above)
Comment (3van) [inline src/org/opensolaris/opengrok/analysis/lua/LuaSymbolTokenizer.lex]: Hah! I thought I grepped for all those. Guess not. :dancer:

**Comment by 3van** [comment]
I haven't had time to work on this recently but I'll at least fix the typo and see if I can get some basic tests done this weekend. Thanks for verifying the OCA; that's good to know as well.

File: src/org/opensolaris/opengrok/analysis/lua/LuaSymbolTokenizer.lex (hunk 1 above)
Comment (tarzanek) [inline src/org/opensolaris/opengrok/analysis/lua/LuaSymbolTokenizer.lex]: grep? pff ... you should have used grok to find all of them! :-D

**Comment by tarzanek** [comment]
@3van any news? :)  I'd like to merge this, so we can move on, we can push tests with new pull req (divide and conquer)

**Comment by tarzanek** [comment]
@3van I am just going to accept this, let's move the tests to a new pull req, I will fix the silly typos for you

**Comment by tarzanek** [comment]
and thank you @3van ! :)

**Pull request merged**